src/netaddress.cpp

```diff
@@ -538,22 +529,33 @@ std::vector<unsigned char> CNetAddr::GetGroup(const std::vector<bool> &asmap) co
     }
 
     // push our ip onto vchRet byte by byte...
+    size_t i = 0;
     while (nBits >= 8)
     {
-        vchRet.push_back(GetByte(15 - nStartByte));
-        nStartByte++;
+        assert(i < m_addr.size());
+        vchRet.push_back(m_addr[i]);
+        ++i;
```

Comment:
It looks like you can do this as a copy operation now that the ordering is the same.
```C++
size_t i = nBits >> 3;
vchRet.insert(vchRet.end(), m_addr.begin(), m_addr.begin() + i);
nBits &= 0x7;
```
(untested and possibly completely broken)

Reply:
I believe that's correct.

Reply:
Done - [diff together with other suggestions](https://github.com/bitcoin/bitcoin/compare/8b7f2c13b67ae84a59f29a7cffe243055a2b93d9..86b2397faedcd9b57649ddb90e98c8cf6a4a6ad3).